code/modules/spells/spell_code.dm

```diff
@@ -108,10 +114,16 @@ var/list/spells = typesof(/spell) //needed for the badmin verb for now
 		return (target in options)
 	return ((target in view_or_range(range, user, selection_type)) && istype(target, /mob/living))
 
-/spell/proc/perform(mob/user = usr, skipcharge = 0) //if recharge is started is important for the trigger spells
+/spell/proc/perform(mob/user, skipcharge = 0, list/target_override) //if recharge is started is important for the trigger spells
+	if(!user)
+		user = usr
+
 	if(!holder)
 		holder = user //just in case
-	if(spell_flags & WAIT_FOR_CLICK)
+
+	var/list/targets = target_override
```

Comment:
If you do this, there's a similar system in aoe_turf that should be replaced that does an identical thing